Akkadu/rsi-widgets
# Feat/retry widget event

## Motivation and Context
The refresh button allows avoiding the issue when `auto-play` fails (especially in safari browser) But this is a embedded solution with its own style and logic.  In this PR we add a custom way to handle the refresh, where the developer can customer when and how to refresh the SDK.

Once the `refreshCallback` is on use, in the first place the default refresh won't be shown, in the second place on require refresh the callback will receive a call with the error message and the function requires to refresh the SDK. (so there is no requires to dig or ref the component). 

In Addition, we have added a manual way to display the refresh button which can help on testing.



## Set up Test
You will need to download the PR and run in each project demo, where the code is ready to test the feature.

> **Vue**
Run Vue: cd packages/rsi-vue yarn start
Run VueDemo: cd packages/rsi-vue/dev yarn start

> **React**
Run Vue: cd packages/rsi-react yarn start
Run VueDemo: cd packages/rsi-react/demo yarn start

> **Vainilla** Run Vue: cd packages/rsi-vanilla http-serve To run a server for static HTML

When changing to a language **which is not floor**, we have enabled show the refresh button to make it easier to test.
After is approved we will comment on this line.

### How to test
Run a specific project.
- Remove `callbackRefresh` property. On changing language default Refresh button shall appear.
- Restore `callbackRefresh` property. Onc hanging language `custom refresh` button shall re-appear.
- Click on the custom refresh button, a log of `unmuting` the language shall appear in console log. Then we had set that the button disappear as the default button logic.


## Screenshots (if appropriate)
VUE: 
![image](https://user-images.githubusercontent.com/37141039/141823764-f243bdcb-7d9b-4d84-a06a-c5c4e4d02a89.png)
Vainilla
![image](https://user-images.githubusercontent.com/37141039/141823973-6ac808b4-9630-4b49-834e-feac4c68351c.png)
React
![image](https://user-images.githubusercontent.com/37141039/141824550-8868003d-f14f-4dbb-8249-c2b79e02c402.png)



## Checklist



加油！
